game-app/game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -2745,4 +2746,158 @@ public Map<String, MutableProperty<?>> getPropertyMap() {
                 this::resetTerritoryEffectProperty))
         .build();
   }
+
+  @Override
+  public Function<String, MutableProperty<?>> getPropertyMapFunction() {
```

Comment:
Another benefit is this method body is shorter than getPropertyMap() above it, so if we do the replacement, we also make the code less verbose.